sdk/dotnet/Pulumi/Resources/StackReference.cs

```diff
@@ -0,0 +1,185 @@
+
+        /// <summary>
+        /// The names of any stack outputs which contain secrets.
+        /// </summary>
+        [Output("secretOutputNames")]
+        public Output<ImmutableArray<string>> SecretOutputNames { get; private set; } = null!;
+
+        /// <summary>
+        /// Create a <see cref="StackReference"/> resource with the given unique name, arguments, and options.
+        /// <para />
+        /// If args is not specified, the name of the referenced stack will be the name of the StackReference resource.
+        /// </summary>
+        /// <param name="name">The unique name of the stack reference.</param>
+        /// <param name="args">The arguments to use to populate this resource's properties.</param>
+        /// <param name="options">A bag of options that control this resource's behavior.</param>
+        public StackReference(string name, StackReferenceArgs? args = null, ResourceOptions? options = null)
+            : base("pulumi:pulumi:StackReference",
+                  name,
+                  new StackReferenceArgs { Name = args?.Name ?? name },
+                  CustomResourceOptions.Merge(options, new CustomResourceOptions { Id = args?.Name ?? name }))
+        {
+        }
+
+        /// <summary>
+        /// Fetches the value of the named stack output, or null if the stack output was not found.
+        /// </summary>
+        /// <param name="name">The name of the stack output to fetch.</param>
+        /// <returns>An <see cref="Output{T}"/> containing the requested value.</returns>
+        public Output<object?> GetOutput(Input<string> name)
+        {
+            // Note that this is subltly different from "Apply" here. A default "Apply" will set the secret bit if any
+            // of the inputs are a secret, and this.Outputs is always a secret if it contains any secrets. We do this dance
+            // so we can ensure that the Output we return is not needlessly tainted as a secret.
+            var inputs = (Input<ImmutableDictionary<string, object>>)this.Outputs;
+            var value = Output.Tuple(name, inputs).Apply(v => v.Item2.ContainsKey(v.Item1) ? v.Item2[v.Item1] : default);
+
+            var isSecret = IsSecretOutputName(name);
+            return new Output<object?>(value.Resources, value.ChangeSecretness(isSecret));
+        }
+
+        /// <summary>
+        /// Fetches the value of the named stack output, or throws an error if the output was not found.
+        /// </summary>
+        /// <param name="name">The name of the stack output to fetch.</param>
+        /// <returns>An <see cref="Output{T}"/> containing the requested value.</returns>
+        public Output<object> RequireOutput(Input<string> name)
+        {
+            var inputs = (Input<ImmutableDictionary<string, object>>)this.Outputs;
+            var value = Output.Tuple(name, inputs).Apply(v => 
+                v.Item2.ContainsKey(v.Item1) 
+                    ? v.Item2[v.Item1]
```

Comment:
nicer to do `v.Item2.TryGetValue(v.Item, out var result) ? result : throw ...`

Reply:
I loath out parameters but I changed it :)